ksqldb-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java

```diff
@@ -88,13 +92,18 @@ public KsqlEngineMetrics(
       final Optional<KsqlMetricsExtension> metricsExtension
   ) {
     this.ksqlEngine = ksqlEngine;
-    this.ksqlServiceId = ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX
+    this.ksqlServiceIdLegacyPrefix = ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX
         + ksqlEngine.getServiceId();
+    this.ksqlServicePrefix = ReservedInternalTopics.CONFLUENT_PREFIX;
     this.sensors = new ArrayList<>();
     this.countMetrics = new ArrayList<>();
     this.metricGroupPrefix = Objects.requireNonNull(metricGroupPrefix, "metricGroupPrefix");
     this.metricGroupName = metricGroupPrefix + METRIC_GROUP_POST_FIX;
     this.customMetricsTags = customMetricsTags;
+
+    final Map<String, String> metricsTags = new HashMap<>(customMetricsTags);
+    metricsTags.put(KsqlConstants.KSQL_SERVICE_ID_METRICS_TAG, ksqlEngine.getServiceId());
+    this.newCustomMetricsTags = ImmutableMap.copyOf(metricsTags);
```

Comment:
Could this be more efficient as it does 1 copy instead of 2?.
```
this.newCustomMetricsTags = ImmutableMap.<String, String>builder()
       .putAll(customMetricsTags)
       .put(KsqlConstants.KSQL_SERVICE_ID_METRICS_TAG, ksqlEngine.getServiceId())
       .build();
```

Reply:
done